### PR TITLE
Pitch: Wrapper that adds human-mode rendering to any environment with rgb_array-rendering

### PR DESCRIPTION
**Pull request by Markus28**
This PR is supposed to provide a wrapper that adds human-rendering functionality to environments that only support rgb_array rendering out of the box

---
### Pitch
Oftentimes I have found myself in a situation where I had implemented an environment that could produce RGB images, but that could not actually render the environment in real-time. It would be convenient to have a wrapper that adds this functionality. It's not that hard to add this feature yourself, but if you don't use PyGame for rendering in the first place, it may be *a lot* of boilerplate code.

---

If someone thinks that this might be a good idea, I will start adding documentation and tests if necessary. Not sure whether I'm missing something, but this seems like a wrapper that could be quite useful (at least I will be using it personally).

### CODE REVIEW

**Comment by jkterry1** [comment]
On hold until new rendering API is merged, also has merge conflicts

**Comment by pseudo-rnd-thoughts** [comment]
Could you add some tests for this?

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
It generally looks good, could add a some tests? More to check that the code runs than to check the render output

File: gym/__init__.py
Comment (Markus28) [reply gym/__init__.py:5]: It was already there, this will hide a welcome message from PyGame. I moved it up because `from gym.envs import make, spec, register` will import gym.wrappers, which will import PyGame in human_rendering.py. So we need to set the environment variable before importing `gym.envs`. The `noqa`s are then required because flake8 will complain otherwise. But I guess this will all be solved by moving PyGame into the Wrapper.

**Comment by Markus28** [comment]
I'm no longer sure whether using a wrapper is actually the best approach. It would have worked well with the old API but the new API leads to two new restrictions:

- We cannot change the render mode to something else when initializing the wrapper (i.e. the wrapper can only use human-rendering)
- The wrapped environment must use rgb_array rendering

Maybe it might be more convenient if we somehow include the functionality in `gym.utils.renderer.Renderer`. People who use `Renderer` wouldn't have to worry about wrapping the environment and the two issues above would be solved.

File: gym/__init__.py
Comment (Markus28) [reply gym/__init__.py:7]: See the other comment

**Comment by pseudo-rnd-thoughts** [comment]
Would you be able to add some tests for this as it affects a fairly critical part of the code
This may be easier once https://github.com/openai/gym/pull/2900 is merged

File: tests/wrappers/test_human_rendering.py
Comment (pseudo-rnd-thoughts) [inline tests/wrappers/test_human_rendering.py:24]: Wrap with `re.escape("...")` to avoid regex issues

File: gym/envs/registration.py
Comment (pseudo-rnd-thoughts) [inline gym/envs/registration.py:627]: Would we want to include the actual error, just in case it is not this error?

File: gym/wrappers/human_rendering.py
Comment (Markus28) [reply gym/wrappers/human_rendering.py:37]: We use it in the _render_frame method in self.clock.tick

File: gym/envs/registration.py
Comment (Markus28) [reply gym/envs/registration.py:627]: I think it should appear in the stacktrace

Comment (pseudo-rnd-thoughts) [reply gym/envs/registration.py:627]: Fair then ignore my comment

**Comment by Markus28** [comment]
Thanks, I changed both things :)

**Pull request merged**
